cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cluster/ClusteredSortAgentScheduler.java

```diff
@@ -154,6 +154,28 @@ public void schedule(Agent agent, AgentExecution agentExecution, ExecutionInstru
   }
 
   @Override
+  public ClusteredSortAgentLock tryLock(Agent agent) {
+    // This score determines how much time the agent had left in the set it's currently in.
+    String releaseScore = agentScore(agent);
```

Comment:
we would only care about score in the waiting set in this case (acquire would fail if the agent is in working)